plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/storage/TrieBasedStore.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule.storage;
+
+import org.apache.commons.collections4.trie.PatriciaTrie;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This is a patricia trie based implementation of AttributeValueStore
+ */
+public class TrieBasedStore implements AttributeValueStore {
```

Comment:
I am wondering if this should again be generic? And, do we need the implementation type in the class name? Maybe `DefaultAttributeStore<K,V> implements AttributeStore<K,V>`?

Reply:
Good Point Ankit! I think we can mention the underlying data structure in the javadoc for this class